packages/members-api/lib/repositories/member.js

```diff
@@ -670,29 +706,68 @@ module.exports = class MemberRepository {
                 id: model.id
             });
 
-            if (model.get('plan_id') !== updated.get('plan_id') || model.get('status') !== updated.get('status')) {
-                const originalMrrDelta = getMRRDelta({interval: model.get('plan_interval'), amount: model.get('plan_amount'), status: model.get('status')});
-                const updatedMrrDelta = getMRRDelta({interval: updated.get('plan_interval'), amount: updated.get('plan_amount'), status: updated.get('status')});
+            if (model.get('plan_id') !== updated.get('plan_id') || model.get('status') !== updated.get('status') || model.get('cancel_at_period_end') !== updated.get('cancel_at_period_end')) {
```

Comment:
Having the new `edit` method with update locks in mind, wouldn't it be safer to use `_previousAttributes` on `updated` instead of using `model` (as that read is locked to the write?)

Reply:
This all happens in one transaction and the `model` is fetched with the `FOR UPDATE` lock too so I don't think it matters - but it may be a little cleaner 🤔 If I end up making further changes to this file I'll update it to use `_previousAttributes`

Reply:
(I would keep the model.get in the future for readability, didn't saw the for update lock)